symphony/bdk/ext/group.py

```diff
@@ -0,0 +1,149 @@
+from symphony.bdk.core.extension import BdkAuthenticationAware, BdkApiClientFactoryAware, BdkExtensionServiceProvider
+from symphony.bdk.core.service.user.user_util import extract_tenant_id
+from symphony.bdk.gen.group_api.group_api import GroupApi
+from symphony.bdk.gen.group_model.add_member import AddMember
+from symphony.bdk.gen.group_model.create_group import CreateGroup
+from symphony.bdk.gen.group_model.group_list import GroupList
+from symphony.bdk.gen.group_model.member import Member
+from symphony.bdk.gen.group_model.read_group import ReadGroup
+from symphony.bdk.gen.group_model.sort_order import SortOrder
+from symphony.bdk.gen.group_model.status import Status
+from symphony.bdk.gen.group_model.update_group import UpdateGroup
+from symphony.bdk.gen.group_model.upload_avatar import UploadAvatar
+from symphony.bdk.gen.login_api.authentication_api import AuthenticationApi
+
+
+class SymphonyGroupBdkExtension(BdkAuthenticationAware, BdkApiClientFactoryAware, BdkExtensionServiceProvider):
+    """ Extension for Symphony Groups APIs
+    """
+    def __init__(self):
+        self._api_client_factory = None
+        self._bot_session = None
+
+    def set_api_client_factory(self, api_client_factory):
+        self._api_client_factory = api_client_factory
+
+    def set_bot_session(self, bot_session):
+        self._bot_session = bot_session
+
+    def get_service(self):
+        return SymphonyGroupService(self._api_client_factory, self._bot_session)
+
+
+class SymphonyGroupService:
+    """ Service class for managing Symphony Groups
+    """
+    def __init__(self, api_client_factory, session):
+        self._api_client_factory = api_client_factory
+        self._session = session
+        self._oauth_session = OAuthSession(self._api_client_factory.get_login_client(), self._session)
+
+        client = self._api_client_factory.get_client("/profile-manager")
+        # to enable setting the authorization header defined in api_client.ApiClient.update_params_for_auth
+        client.configuration.auth_settings = lambda: {
+            "bearerAuth": {"in": "header", "type": "bearer", "key": "Authorization",
+                           "value": "Bearer " + self._oauth_session.bearer_token()}}
+        # needed for the x_symphony_host parameter to allow empty value
+        client.configuration._disabled_client_side_validations = ["minLength"]
+        self._group_api = GroupApi(client)
+
+    async def insert_group(self, create_group: CreateGroup) -> ReadGroup:
```

Comment:
I was going to say we should name it create group to be consistent with other services but then in the Java BDK it is named like that, I missed it during the Java BDK review :(